fe/src/main/java/org/apache/doris/catalog/Catalog.java

```diff
@@ -1177,6 +1190,10 @@ private void startMasterOnlyDaemonThreads() {
         // start routine load scheduler
         routineLoadScheduler.start();
         routineLoadTaskScheduler.start();
+        // start dynamic partition task
+        if (Config.dynamic_partition_enable) {
+            dynamicPartitionScheduler.start();
```

Comment:
you make the config `dynamic_partition_enable` mutable at runtime, but here you do not start the thread if `dynamic_partition_enable` is false when FE starting up. Then even if you change this config to true at runtime, the thread will not start.

You should just let this thread start, but check the `dynamic_partition_enable` at every loop of the dynamic partition scheduling.